### PR TITLE
SAAS-2293: Plugin to check Elasticsearch use dedicated master nodes.

### PR DESCRIPTION
**Pull request by UmairAjmal**

### CODE REVIEW

File: plugins/aws/es/esDedicatedMasterEnabled.js [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+            }
+
+            if (!listDomainNames.data.length){
+                helpers.addResult(results, 0, 'No ES domains found', region);
+                return rcb();
+            }
+
+            listDomainNames.data.forEach(function(domain){
+                var describeElasticsearchDomain = helpers.addSource(cache, source,
+                    ['es', 'describeElasticsearchDomain', region, domain.DomainName]);
```
Comment (AkhtarAmir) [inline plugins/aws/es/esDedicatedMasterEnabled.js:38]:
```suggestion
            listDomainNames.data.forEach(function(domain){
                if (!domain.DomainName) return;
                var describeElasticsearchDomain = helpers.addSource(cache, source,
                    ['es', 'describeElasticsearchDomain', region, domain.DomainName]);
```

File: plugins/aws/es/esDedicatedMasterEnabled.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+                    describeElasticsearchDomain.err ||
+                    !describeElasticsearchDomain.data ||
+                    !describeElasticsearchDomain.data.DomainStatus) {
+                    helpers.addResult(
+                        results, 3,
+                        'Unable to query for ES domain config: ' + helpers.addError(describeElasticsearchDomain), region);
+                } else {
+                    var localDomain = describeElasticsearchDomain.data.DomainStatus;
+
+                    if (localDomain.ElasticsearchClusterConfig &&
```
Comment (AkhtarAmir) [inline plugins/aws/es/esDedicatedMasterEnabled.js:51]:
```suggestion
                    if (localDomain && localDomain.ElasticsearchClusterConfig &&
```

File: plugins/aws/es/esDedicatedMasterEnabled.js [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+            }
+
+            if (!listDomainNames.data.length){
+                helpers.addResult(results, 0, 'No ES domains found', region);
+                return rcb();
+            }
+
+            listDomainNames.data.forEach(function(domain){
+                if (!domain.DomainName) return;
+                var describeElasticsearchDomain = helpers.addSource(cache, source,
```
Comment (AkhtarAmir) [inline plugins/aws/es/esDedicatedMasterEnabled.js:38]: Create ARN

Comment (AkhtarAmir) [inline plugins/aws/es/esDedicatedMasterEnabled.js:38]: Create ARN, include in unknown check and use that ARN in results

**Review by AkhtarAmir — CHANGES_REQUESTED** [review]

**Review by AkhtarAmir — APPROVED** [review]

File: plugins/aws/es/esDedicatedMasterEnabled.js [hunk 4]
```diff
@@ -0,0 +1,71 @@
+var async = require('async');
+var helpers = require('../../../helpers/aws');
+
+module.exports = {
+    title: 'ElasticSearch Dedicated Master Enabled',
+    category: 'ES',
+    description: 'Ensure that Amazon Elasticsearch clusters are using dedicated master nodes.',
```
Comment (AkhtarAmir) [inline plugins/aws/es/esDedicatedMasterEnabled.js:7]:
```suggestion
    description: 'Ensure that Amazon Elasticsearch domains are using dedicated master nodes.',
```

**Review by AkhtarAmir — APPROVED** [review]

**Review by giorod3 — APPROVED** [review]

**Pull request merged**